src/keystore/sdcard.py

```diff
@@ -66,24 +73,41 @@ async def save_mnemonic(self, path=None):
             if not platform.is_sd_present():
                 raise KeyStoreError("SD card is not present")
             platform.mount_sdcard()
-        self.save_aead(path, plaintext=self.mnemonic.encode(), key=self.enc_secret)
+
+        if platform.file_exists("%s/%s.%s" % (path, self.fileprefix(path), filename)):
+            scr = Prompt(
+                "\n\nFile already exists: %s\n" % filename,
+                "Would you like to overwrite this file?",
+            )
+            res = await self.show(scr)
+            if res is False:
+                return None
+
+        self.save_aead("%s/%s.%s" % (path, self.fileprefix(path), filename), plaintext=self.mnemonic.encode(),
+                       key=self.enc_secret)
         if path == self.sdpath:
             platform.unmount_sdcard()
         # check it's ok
-        await self.load_mnemonic(path)
+        await self.load_mnemonic(path, "%s.%s" % (self.fileprefix(path), filename))
         return True
 
     @property
     def is_key_saved(self):
-        flash_exists = platform.file_exists(self.flashpath)
+        flash_files = sum(
+            [[f[0] for f in os.ilistdir(self.flashpath) if f[0].lower().startswith(self.fileprefix(self.flashpath))]],
+            [])
+        flash_exists = False if len(flash_files) == 0 else True
```

Comment:
Not sure I understand why you need `sum` here. Also `flash_exists` check can be simplified.
```suggestion
        flash_files = [
            f[0] for f in os.ilistdir(self.flashpath)
            if f[0].lower().startswith(self.fileprefix(self.flashpath))
        ]
        flash_exists = (len(flash_files) > 0)
```